src/nss/new_nss_oslogin.c

```diff
@@ -235,9 +239,189 @@ int recvline(struct Buffer *const buffer) {
     buffer->buflen += recvlen;
 
     if (recvbuf[recvlen - 1] == '\n') {
+      free(recvbuf);
       return buffer->buflen;
     }
   }
 
+  free(recvbuf);
   return -1;  // Unreachable code.
 }
+
+static enum nss_status
+_nss_oslogin_getpwnam_r(const char *name, struct passwd *result, char *buffer,
+                        size_t buflen, int *errnop) {
+  int res;
+  struct Buffer mgr;
+  memset(&mgr, 0, sizeof(struct Buffer));
+
+  *errnop = 0;
+
+  if (dial(&mgr) != 0) {
```

Comment:
Do you want to set errno for failures in dial/send? As it is here, the behavior is identical to receiving an empty response. That might be what you want, but just want to make sure you've considered it. 

https://www.gnu.org/software/libc/manual/html_node/NSS-Modules-Interface.html